pip/wheel.py

```diff
@@ -692,6 +697,18 @@ def __build_one(self, req, tempd):
             logger.error('Failed building wheel for %s', req.name)
             return False
 
+    def _clean_one(self, req):
+        base_args = self._base_setup_args(req)
+
+        logger.info('Running setup.py clean for %s', req.name)
+        clean_args = base_args + ['clean', '--all']
+        try:
+            call_subprocess(clean_args, cwd=req.source_dir, show_stdout=False)
+            return True
+        except:
+            logger.error('Failed cleaning build dir for %s', req.name)
```

Comment:
But the install could be broken. We can't tell. I couldn't find a good definition of what the different levels actually mean, but my instinct says that warning means "you should know this but everything is basically OK" and error says "you need to know this, something's wrong". There's nothing that covers "something *might* be wrong" but I'm inclined to think that by the time you get here a wheel build has failed, and now you can't clean up, so things seem to be going badly and an error is probably warranted.